src/anomalib/models/image/supersimplenet/lightning_model.py

```diff
@@ -0,0 +1,207 @@
+
+        self.model = SupersimplenetModel(
+            perlin_threshold=perlin_threshold,
+            backbone=backbone,
+            layers=layers,
+            stop_grad=stop_grad,
+        )
+        self.loss = SSNLoss()
+
+    def training_step(self, batch: Batch, *args, **kwargs) -> STEP_OUTPUT:
+        """Perform the training step input and return the loss.
+
+        Args:
+            batch (batch: dict[str, str | torch.Tensor]): Input batch
+            args: Additional arguments.
+            kwargs: Additional keyword arguments.
+
+        Returns:
+            STEP_OUTPUT: Dictionary containing the loss value.
+        """
+        del args, kwargs  # These variables are not used.
+
+        anomaly_map, anomaly_score, masks, labels = self.model(
+            images=batch.image,
+            masks=batch.gt_mask,
+            labels=batch.gt_label,
+        )
+        loss = self.loss(pred_map=anomaly_map, pred_score=anomaly_score, target_mask=masks, target_label=labels)
+        self.log("train_loss", loss.item(), on_epoch=True, prog_bar=True, logger=True)
+        return {"loss": loss}
+
+    def validation_step(self, batch: Batch, *args, **kwargs) -> STEP_OUTPUT:
+        """Perform the validation step and return the anomaly map and anomaly score.
+
+        Args:
+            batch (dict[str, str | torch.Tensor]): Input batch
+            args: Additional arguments.
+            kwargs: Additional keyword arguments.
+
+        Returns:
+            STEP_OUTPUT | None: batch dictionary containing anomaly-maps.
+        """
+        # These variables are not used.
+        del args, kwargs
+
+        # Get anomaly maps and predicted scores from the model.
+        predictions = self.model(batch.image)
+
+        return batch.update(**predictions._asdict())
+
+    @property
+    def trainer_arguments(self) -> dict[str, Any]:
+        """Return SuperSimpleNet trainer arguments."""
+        return {"gradient_clip_val": self.norm_clip_val, "num_sanity_val_steps": 0}
+
+    def configure_optimizers(self) -> OptimizerLRScheduler:
+        """Configure AdamW optimizer and MultiStepLR scheduler."""
+        optim = AdamW(
+            [
+                {
+                    "params": self.model.adaptor.parameters(),
+                    "lr": 0.0001,
+                },
+                {
+                    "params": self.model.segdec.parameters(),
+                    "lr": 0.0002,
+                    "weight_decay": 0.00001,
+                },
+            ],
+        )
+        sched = MultiStepLR(
+            optim,
+            milestones=[int(self.trainer.max_epochs * 0.8), int(self.trainer.max_epochs * 0.9)],
+            gamma=0.4,
+        )
+        return [optim], [sched]
+
+    @property
+    def learning_type(self) -> LearningType:
+        """Return the learning type of the model.
+
+        Returns:
+            LearningType: Learning type of the model.
+        """
+        return LearningType.ONE_CLASS
```

Comment:
I guess this would change when we add support for the supervised mode to the overall Anomalib pipeline. Maybe you could add a comment about this, as it might confuse users. Just a brief mention in the docstring would be sufficient.